### PR TITLE
26/reset button

### LINKED ISSUE #22: Add Grid toggle

### LINKED ISSUE #23: Add Emitted Lines toggle

### LINKED ISSUE #24: Rename Spectrum Bar as Wavelength distribution

### LINKED ISSUE #25: Decrease Thermometer height

### LINKED ISSUE #26: Add reset button

### PR DESCRIPTION
**Pull request by pyphilia**
close #26 
close #25 
close #24 
close #23 
close #22 

and I changed 100nm to 5000nm 
We could have a simpler unit but since the wavelengths distribution is in nm as well I let it be. Maybe they will ask for a refactor.

### CODE REVIEW

File: src/actions/lab.js [hunk 1]
```diff
@@ -29,9 +34,45 @@ const setShowThermometerLabels = (showThermometerLabels) => (dispatch) =>
...
+const toggleElectrons = (payload) => (dispatch) => {
+  dispatch({
+    type: TOGGLE_ELECTRONS,
+    payload,
+  });
+};
+
+const toggleWavelengthDistribution = (payload) => (dispatch) => {
+  dispatch({
+    type: TOGGLE_SPECTRUM_BAR,
```
Comment (hagoptaminian) [inline src/actions/lab.js:58]: Rename this to `TOGGLE_WAVELENGTH_DISTRIBUTION` here and elsewhere - makes it easier to maintain later on if someone does't know that the spectrum_bar is the same thing as the wavelength_distribution

**Review by hagoptaminian — APPROVED** [review]
looks great! left a few small comments... and a few other notes:

- when the SideMenu is toggled off, the grass doesn't extend to the width of the screen.
- like you suggested we can wait for feedback but IMO a grid-square-width of `47.5 / 2.5` (i.e. each square is 2000nm) makes measurement easier on the eye
- and another thought: on the wavelength distribution, I find the scale jumps of 3,250 a bit odd, and that they make measurement harder. for example, at the default temperature, the red line is somewhere between 6,250 and 9,500, and (1) these are not easy numbers to work with, (2) it's not clear where each number is positioned. so (1) why not just 3,000? (2) would be nice if there were little ticks on the x-axis to make it easier to measure things...

**Comment by pyphilia** [comment]
Thank you @hagoptaminian for your reviews! I updated the branch accordingly + the feedback from Phillipe.

close #28

File: src/components/common/SideMenu.js [hunk 2]
```diff
@@ -119,6 +135,20 @@ class SideMenu extends React.Component {
     dispatchSetIsPaused(!isPaused);
   };
 
+  reset = () => {
+    const { dispatchResetSettings } = this.props;
+    dispatchResetSettings();
+  };
+
+  onScaleUnitToggle = () => {
```
Comment (juancarlosfarah) [inline src/components/common/SideMenu.js:143]: `onToggleScaleUnit`?

File: src/components/common/SideMenu.js [hunk 3]
```diff
@@ -119,6 +135,20 @@ class SideMenu extends React.Component {
...
+    const { dispatchSetScaleUnit, showKelvinScale } = this.props;
+    if (showKelvinScale) {
```
Comment (juancarlosfarah) [inline src/components/common/SideMenu.js:145]: Shouldn't this be the opposite? If it's not the verb but the boolean, then maybe rename `showKelvinScale: currentlyShowingKelvinScale` or something like that.

File: src/components/lab/WavelengthDistribution.js [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+const getSpectralPowerDensityAt = ({ wavelength, temperature }) => {
+  // Avoiding division by 0
+  if (wavelength === 0) {
+    return 0;
+  }
+
+  const B = 1.43877e7; // is hc/k in units of nanometer-kelvin
+  const c = SPEED_OF_LIGHT_CONSTANT;
+  const h = PLANCK_CONSTANT;
+  return (
```
Comment (juancarlosfarah) [inline src/components/lab/WavelengthDistribution.js:42]: Maybe explain this formula? Or at least just mention it with a link?

File: src/components/lab/WavelengthDistribution.js [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    WAVELENGTH_DISTRIBUTION_MAX_WAVELENGTH -
+    WAVELENGTH_DISTRIBUTION_MIN_WAVELENGTH;
+
+  // centers spectrum bar horizontally
+  const wavelengthDistributionInitialXPosition =
+    stageWidth -
+    WAVELENGTH_DISTRIBUTION_WIDTH -
+    WAVELENGTH_DISTRIBUTION_MARGIN -
+    2 * WAVELENGTH_DISTRIBUTION_PADDING;
+  const wavelengthDistributionInitialYPosition = 0.12 * stageHeight;
```
Comment (juancarlosfarah) [inline src/components/lab/WavelengthDistribution.js:62]: Why `0.12`?

File: src/components/lab/thermometer/Scale.js [hunk 6]
```diff
@@ -229,21 +232,33 @@ const Scale = ({
...
         }
-        text={t(name)}
-        fontSize={SCALE_FONT_SIZE}
-      />
+      >
+        <Line
+          x={-THERMOMETER_WIDTH}
+          points={[0, 0, THERMOMETER_WIDTH + SCALE_TEXT_WIDTH_FACTOR - 5, 0]}
+          stroke={SCALE_TICKS_STROKE_COLOR}
+          dash={[6, 2]}
```
Comment (juancarlosfarah) [inline src/components/lab/thermometer/Scale.js:248]: What are these numbers for dash? Can we explain them or refactor them?

File: src/config/constants.js [hunk 7]
```diff
@@ -109,11 +115,11 @@ export const GROUND_FILL_DARK_COLOR = '#ad7e4e';
...
+export const GROUND_GRASS_BACKGROUND = '#a14e06';
 
 // grid constants
 export const GRID_AXES_COLOR = '#000';
 export const GRID_AXES_STROKE_WIDTH = 0.5;
-export const GRID_SQUARE_WIDTH_AND_HEIGHT = 47.5;
-export const GRID_LEGEND_LABEL_TEXT = '100nm';
-export const GRID_UNIT_SQUARE_LENGTH = 45;
+export const GRID_SQUARE_WIDTH_AND_HEIGHT = 33;
+export const GRID_LEGEND_LABEL_TEXT = '1000nm = 1um';
```
Comment (juancarlosfarah) [inline src/config/constants.js:124]: You can use `µ` instead of `u` I believe.

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Really solid PR. :atom: 

Some minor comments and questions.

File: src/components/lab/WavelengthDistribution.js (hunk 5 above)
Comment (pyphilia) [reply src/components/lab/WavelengthDistribution.js:62]: Because of the header, I'll check if I can rather depend on its height.

**Review by hagoptaminian — APPROVED** [review]
Really cool. Love the addition of 'Most Frequent' on the wavelength distribution chart. Nicely done!

**Pull request merged**